unixfs/src/pb/unixfs.rs

```diff
@@ -0,0 +1,229 @@
+    fn write_message<W: WriterBackend>(&self, w: &mut Writer<W>) -> Result<()> {
+        w.write_with_tag(8, |w| w.write_enum(*&self.Type as i32))?;
+        if let Some(ref s) = self.Data {
+            w.write_with_tag(18, |w| w.write_bytes(&**s))?;
+        }
+        if let Some(ref s) = self.filesize {
+            w.write_with_tag(24, |w| w.write_uint64(*s))?;
+        }
+        for s in &self.blocksizes {
+            w.write_with_tag(32, |w| w.write_uint64(*s))?;
+        }
+        if let Some(ref s) = self.hashType {
+            w.write_with_tag(40, |w| w.write_uint64(*s))?;
+        }
+        if let Some(ref s) = self.fanout {
+            w.write_with_tag(48, |w| w.write_uint64(*s))?;
+        }
+        if let Some(ref s) = self.mode {
+            w.write_with_tag(56, |w| w.write_uint32(*s))?;
+        }
+        if let Some(ref s) = self.mtime {
+            w.write_with_tag(66, |w| w.write_message(s))?;
+        }
+        Ok(())
+    }
+}
+pub mod mod_Data {
+    #[derive(Debug, PartialEq, Eq, Clone, Copy)]
+    pub enum DataType {
+        Raw = 0,
+        Directory = 1,
+        File = 2,
+        Metadata = 3,
+        Symlink = 4,
+        HAMTShard = 5,
+    }
+    impl Default for DataType {
+        fn default() -> Self {
+            DataType::Raw
+        }
+    }
+    impl From<i32> for DataType {
+        fn from(i: i32) -> Self {
+            match i {
+                0 => DataType::Raw,
+                1 => DataType::Directory,
+                2 => DataType::File,
+                3 => DataType::Metadata,
+                4 => DataType::Symlink,
+                5 => DataType::HAMTShard,
+                _ => Self::default(),
+            }
+        }
+    }
+    impl<'a> From<&'a str> for DataType {
+        fn from(s: &'a str) -> Self {
+            match s {
+                "Raw" => DataType::Raw,
+                "Directory" => DataType::Directory,
+                "File" => DataType::File,
+                "Metadata" => DataType::Metadata,
+                "Symlink" => DataType::Symlink,
+                "HAMTShard" => DataType::HAMTShard,
+                _ => Self::default(),
+            }
+        }
+    }
+    impl From<DataType> for i32 {
+        fn from(dt: DataType) -> Self {
+            match dt {
+                DataType::Raw => 0,
+                DataType::Directory => 1,
+                DataType::File => 2,
+                DataType::Metadata => 3,
+                DataType::Symlink => 4,
+                DataType::HAMTShard => 5,
+            }
+        }
+    }
+}
+#[derive(Debug, Default, PartialEq, Clone)]
+pub struct UnixTime {
+    pub Seconds: i64,
+    pub FractionalNanoseconds: Option<u32>,
+}
+impl<'a> MessageRead<'a> for UnixTime {
+    fn from_reader(r: &mut BytesReader, bytes: &'a [u8]) -> Result<Self> {
+        let mut msg = Self::default();
+        while !r.is_eof() {
+            match r.next_tag(bytes) {
+                Ok(8) => msg.Seconds = r.read_int64(bytes)?,
```

Comment:
nit(also applies to other spots like this): these numbers could use some nice `const`s indicating their significance

Reply:
The tags are from the proto file (the field identifiers). I guess the codegen could be made better but these are rarely read, except maybe to learn about protobuf decoding, if one happens to start yet another protobuf decoding library ... :D. I did generate this with the UB causing Owned wrapper types, deleted them manually and fixed up some clippy nits and forgot the `--owned` it to the generation script.

Reply:
Removed.